src/pages/staticPages/tools/oncoprinter/Oncoprinter.tsx

```diff
@@ -379,9 +397,177 @@ export default class Oncoprinter extends React.Component<
         return WindowStore.size.width - 75;
     }
 
+    @action.bound
+    public handleSelectedClinicalTrackColorChange(
+        value: string,
+        color: RGBAColor | undefined
+    ) {
+        if (this.selectedClinicalTrack) {
+            this.props.store.setUserSelectedClinicalTrackColor(
+                this.selectedClinicalTrack.label,
+                value,
+                color
+            );
+        }
+    }
+
+    @observable trackKeySelectedForEdit: string | null = null;
+
+    @action.bound
+    setTrackKeySelectedForEdit(key: string | null) {
+        this.trackKeySelectedForEdit = key;
+    }
+
+    @computed get selectedClinicalTrack() {
+        return _.find(
+            this.props.store.clinicalTracks,
+            t => t.key === this.trackKeySelectedForEdit
+        );
+    }
+
+    @computed get defaultClinicalAttributeColoringForStringDatatype() {
```

Comment:
seems odd to me that the defaults would be derived from selected, no?  maybe i don't quite understand

Reply:
the defaults are obtained from `RESERVED_CLINICAL_VALUE_COLORS` but we need the selected track data to get the values not present in `RESERVED_CLINICAL_VALUE_COLORS` so we can assign them their default colors.